### PR TITLE
Normalize backend name during compute init

### PR DESCRIPTION
**Pull request by phamann**
### TL;DR
To reduce on-boarding friction `compute init` only asks for a backend address during the interactive initialisation, we then use the same value for both the `address` and `name` fields in the [API call](https://docs.fastly.com/api/config#backend_85c170418ee71191dbb3b5046aeb6c2c) to create the backend. However, we have since discovered these are not valid backend `name` values and therefore we need to normalize the address input before creation. 

### Notes
Backend names must match this regular expression (`^[a-zA-Z][a-zA-Z0-9_]*$`), i.e. they myst conform to these rules:
- cannot start with a number.
- can only contain alpha numeric ASCII characters or underscores.

If the input address is an IP address and therefore starts with a number, we prefix the string with "F_" which follows historic naming convention for backends in Fastly VCL services.

### CODE REVIEW

**Comment by phamann** [comment]
>If the normalized name is still invalid for whatever reason, the CreateBackend call will fail, correct?

Sadly this is not the case and I've raised with the downstream API team (cc @thommahoney)

**Comment by phamann** [comment]
Closing as this issue has been resolved upstream.
